### PR TITLE
Add YOLO-World forward and inference

### PR DESCRIPTION
**Pull request by InsaneOnion**
Add YOLO-World forward and inference
some examples of forward alignment：
1. bus.jpg
**torch commands**: python image_demo.py configs/pretrain/yolo_world_v2_x_vlpan_bn_2e-3_100e_4x8gpus_obj365v1_goldg_train_lvis_minival.py /home/onion/workspace/code/PaddleMIX/ppdiffusers/examples/YOLO-World/pretrain/yolo_world_v2_x_obj365v1_goldg_cc3mlite_pretrain-8698fbfa.pth /home/onion/
bus.jpg 'person,bus' --topk 100 --threshold 0.001 --output-dir /home/onion/out/
**paddle commands**: python infer.py --config configs/yolo_world_x.yml -o weights=./pretrain/yolo_world_v2_x_obj365v1_goldg_cc3mlite_pretrain-8698fbfa.pdparams --image /home/onion/bus.jpg --text 'person,bus' --topk 100 --threshold 0.01 --output_dir "/home/onion/out"

2. zidane.jpg
**torch commands**: python image_demo.py configs/pretrain/yolo_world_v2_x_vlpan_bn_2e-3_100e_4x8gpus_obj365v1_goldg_train_lvis_minival.py /home/onion/workspace/code/PaddleMIX/ppdiffusers/examples/YOLO-World/pretrain/yolo_world_v2_x_obj365v1_goldg_cc3mlite_pretrain-8698fbfa.pth /home/onion/
zidane.jpg 'bald man,white haired man' --topk 100 --threshold 0.001 --output-dir /home/onion/out/
**paddle commands**: python infer.py --config configs/yolo_world_x.yml -o weights=./pretrain/yolo_world_v2_x_obj365v1_goldg_cc3mlite_pretrain-8698fbfa.pdparams --image /home/onion/zidane.jpg --text 'bald man,white haired man' --topk 100 --threshold 0.001 --output_dir "/home/onion/out"

### CODE REVIEW

File: ppdiffusers/examples/YOLO-World/README.md [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+```
+
+
+## 3. 下载模型
+通过 [Huggingface](https://huggingface.co/wondervictor/YOLO-World/) 下载YOLO-World 的模型权重文件。可以将权重文件放在 `pretrain/` 目录下，下载完成后执行torch2paddle.py脚本将权重转化为paddle格式。
+
+```bash
+python pretrain/torch2paddle.py <path_to_weight.pth> -p <path_to_weight.pdparams>
+```
+
```
Comment (LokeZhou) [inline ppdiffusers/examples/YOLO-World/README.md]: 可以把转后的权重分享，这边上传自动下载，省略这一步

File: ppdiffusers/examples/YOLO-World/README.md [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+通过 `git clone` 命令拉取 PaddleMIX 源码，并安装必要的依赖库。请确保你的 PaddlePaddle 框架版本在 2.6.0 之后，PaddlePaddle 框架安装可参考 [飞桨官网-安装](https://www.paddlepaddle.org.cn/install/quick?docurl=/documentation/docs/zh/install/pip/linux-pip.html)。
+
+```bash
+# 克隆 PaddleMIX 仓库
+git clone https://github.com/PaddlePaddle/PaddleMIX
+
+# 安装2.6.0版本的paddlepaddle-gpu，当前我们选择了cuda12.0的版本，可以查看 https://www.paddlepaddle.org.cn/ 寻找自己适合的版本
+python -m pip install paddlepaddle-gpu==2.6.0.post120 -f https://www.paddlepaddle.org.cn/whl/linux/mkl/avx/stable.html
+
+# 进入目录
```
Comment (LokeZhou) [inline ppdiffusers/examples/YOLO-World/README.md]: 是不是少了一步pyproject里面的包安装？

File: ppdiffusers/examples/YOLO-World/infer.py [hunk 3]
```diff
@@ -0,0 +1,217 @@
...
+# limitations under the License.
+
+import itertools
+import os
+import os.path as osp
+
+import cv2
+import paddle.nn as nn
+import supervision as sv
+from ppdet.core.workspace import create, load_config, merge_config
```
Comment (LokeZhou) [inline ppdiffusers/examples/YOLO-World/infer.py]: supervision是否第三方库？如果是需要写一个requirments.txt

File: ppdiffusers/examples/YOLO-World/README.md [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+    --config <path_to_config_yaml> \
+    -o weights=<path_to_pretrained_weights> \
+    --image=<path_to_image_or_directory> \
+    --text <text or .txt file path>\
+    --topk=<number_of_top_predictions> \
+    --threshold=<confidence_threshold_value> \
+    --output_dir=<output_directory_path> \
+    --offline
+```
+
```
Comment (LokeZhou) [inline ppdiffusers/examples/YOLO-World/README.md]: 说明一下每个参数

**Comment by LokeZhou** [comment]
整体文件移到PaddleMIX/paddlemix/examples下

File: ppdiffusers/examples/YOLO-World/README.md (hunk 4 above)
Comment (InsaneOnion) [reply ppdiffusers/examples/YOLO-World/README.md]: 好的～

File: ppdiffusers/examples/YOLO-World/README.md (hunk 2 above)
Comment (InsaneOnion) [reply ppdiffusers/examples/YOLO-World/README.md]: 在30行 有提示使用 pip install -e . 安装pyproject中依赖

File: ppdiffusers/examples/YOLO-World/infer.py (hunk 3 above)
Comment (InsaneOnion) [reply ppdiffusers/examples/YOLO-World/infer.py]: supervision是第三方库，但已经作为依赖添加到了pyproject中，还需要另写一个requirments.txt吗？

File: ppdiffusers/examples/YOLO-World/README.md (hunk 1 above)
Comment (InsaneOnion) [reply ppdiffusers/examples/YOLO-World/README.md]: 好的, 已在README中修改～

File: ppdiffusers/examples/YOLO-World/infer.py (hunk 3 above)
Comment (LokeZhou) [reply ppdiffusers/examples/YOLO-World/infer.py]: ok, pyproject有了就行

**Review by LokeZhou — APPROVED** [review]

**Pull request merged**